evap/staff/templates/staff_questionnaire_form.html

```diff
@@ -89,19 +89,27 @@ <h5 class="card-title">{% trans 'Questions' %}</h5>
             rowAdded = function(row) {
                 row.find('a.btn-secondary').remove();
                 applyTomSelect(row.find("select").get());
+                registerSelectChangedHandlers();
             };
             makeFormSortable("question_table", "questions", rowChanged, rowAdded, "", true, true);
 
-            // enable or disable checkbox for additional text answers
-            $('.question-type select').on('select2:select', function (e) {
+            const selectChangedHandler = function(e) {
                 const checkbox = $($(this).closest("td.question-type")).find("input");
-                if (e.params.data.id == 0 || e.params.data.id == 5) {  // 0: Text question; 5: Heading
+                if (e.target.value == 0 || e.target.value == 5) {  // 0: Text question; 5: Heading
```

Comment:
(We talked about this in person)

Although `e.target` should always equal `selectElement`, but let's be sure. Some options
- Currying: `const selectChangeHandler = selectElement => event => { ... }`.although I normally like this approach best, we need to store the curried functions to keep their identity
- Using `this` in a `function` (not arrow function). Yeah, let's not do `this`
- Using `event.currentTarget`. According to mdn, this is equivalent to `this` without the complicated rules of `this` and allows for arrow functions - so a refactor would not break this 

What do you think?